### PR TITLE
Add Metadata Columns in DB

### LINKED ISSUE #795: User-defined names for expansion sets
winner
https://jira.jpl.nasa.gov/browse/AERIEQS-276
Moderate
It is difficult to remember which number means what for each expansion set. It is possible to click through on the editor and see which activities are assigned to a set, but that doesn't help the user when they are on the expansion tab of the plan page. It would be very useful to be able to define a name for each set which would display instead of "Expansion Set X" in most areas of the UI.

### LINKED ISSUE #854: Add extended metadata fields to database
Mission Model: Owner, Date Created, description
Plan: Date Created, Date Updated, Last Updated By, Owner
Simulation Dataset: Owner, Date Created
Expansion Rules: Owner, Date Created, Date Updated, Last Updated By, Description, Subsystem, Category
Expansion Sets: Name, Owner, Date Created, Date Updated, Last Updated By, Description, Subsystem
Sequences: Authored by, Date Created, Date Updated, Version, Subsystem
Constraints: Owner, Date Created, Date Updated, Last Updated By, Description, Subsystem, Category
Scheduling Goals: Owner, Date Created, Description, Subsystem, Category
Scheduling Requests: Scheduler

### LINKED ISSUE #857: Fix GraphQL Schema Issue
We currently track the `scheduling_analysis` table in Hasura. Because this table only has one autogenerated column, the input object Hasura generates has an invalid schema [according to the GraphQL specification](https://spec.graphql.org/October2021/#sel-EAHhBXEBAAAEDFA5FoO).

Solutions:
- Add metadata to `scheduling_analysis`
- Don't track `scheduling_analysis` in Hasura
- Inline `scheduling_analysis` into `scheduling_request`

### PR DESCRIPTION
**Pull request by Mythicaeda**
* **Tickets addressed:** Closes #854, #857, and #795.
* **Review:** By commit  
* **Merge strategy:** Merge (no squash)  

## Description

The first three commits adds metadata columns to tables throughout our databases. The fourth commit addresses #857.

Two things of note:
1. Plans have one owner and multiple collaborators.
2. `activity_types` gained the metadata `subsystem`, which is a nullable text column.

## Verification

No tests need to be updated, as no functionality changed.

Verification that #857 has been resolved was completed by manually checking via the steps defined [here](https://github.com/NASA-AMMOS/aerie/issues/857#issuecomment-1505964717).

## Documentation

No documentation should need updating.

## Future work

Adding the `subsystem` column should help with filtering activities in the UI (and potentially in API queries), although work would need to be done in the backend to allow mission models to specify activity type subsystems. Maybe another Annotation, `@Subsystem("subsystem")`?

### CODE REVIEW

**Comment by camargo** [comment]
We should take advantage of the work @mattdailis did [here](https://jpl.slack.com/archives/CJ07FGRPH/p1681314813727559) for getting mission model metadata into the DB.

**Review by camargo — APPROVED** [review]
I'm good with these changes, just noting we may need additional changes after the tags design is complete. We will make follow up tickets as needed. LGTM.

File: merlin-server/sql/merlin/tables/activity_type.sql
Comment (mattdailis) [inline merlin-server/sql/merlin/tables/activity_type.sql]: 🔻 I'd like to double check with @bashbash - I know we had discussed using the term `category` to be more generic than `subsystem`, and I'm unclear on what decision was made about that.

File: merlin-server/sql/merlin/tables/mission_model.sql
Comment (mattdailis) [inline merlin-server/sql/merlin/tables/mission_model.sql]: 🔶 I'm curious about the need for this default description - it doesn't seem to add any new information, and costs us another trigger to maintain. Could we consider making this a column with a default empty string value?

File: scheduler-server/sql/scheduler/tables/scheduling_request.sql
Comment (mattdailis) [inline scheduler-server/sql/scheduler/tables/scheduling_request.sql]: 🔻 I don't think we've fleshed out the design for this feature enough to the point that I would feel comfortable committing to this implementation. Let's hold off on adding the `requested_scheduler` column.

**Review by mattdailis — CHANGES_REQUESTED** [review]

File: merlin-server/sql/merlin/tables/activity_type.sql
Comment (Mythicaeda) [reply merlin-server/sql/merlin/tables/activity_type.sql]: Activity Types specifically have a subsystem (or potentially several, but @bashbash suggested we start with one) rather than a category.

File: scheduler-server/sql/scheduler/tables/scheduling_request.sql
Comment (Mythicaeda) [reply scheduler-server/sql/scheduler/tables/scheduling_request.sql]: To confirm, are you proposing we keep the removal of `scheduling_analysis` and revisit the topic of how to represent scheduling requests to external schedulers at some point in the future?

Comment (mattdailis) [reply scheduler-server/sql/scheduler/tables/scheduling_request.sql]: Yes, your interpretation is accurate :+1: The idea of an external scheduler listening to and reacting to inserts into `scheduling_request` needs some more design - I think it's too tightly coupled to our representation of scheduling specifications to really be useful to an external scheduler.

I've come around to viewing inlining scheduling analysis into scheduling request as the least unsatisfactory approach to addressing #857 :+1:

File: merlin-server/sql/merlin/tables/activity_type.sql
Comment (mattdailis) [reply merlin-server/sql/merlin/tables/activity_type.sql]: Could you provide a reference to where that decision was documented? I'd like to get product owner sign-off on this prior to merge

File: scheduler-server/sql/scheduler/tables/scheduling_request.sql
Comment (Mythicaeda) [reply scheduler-server/sql/scheduler/tables/scheduling_request.sql]: Done

File: merlin-server/sql/merlin/tables/mission_model.sql
Comment (Mythicaeda) [reply merlin-server/sql/merlin/tables/mission_model.sql]: Done

File: merlin-server/sql/merlin/tables/activity_type.sql
Comment (camargo) [reply merlin-server/sql/merlin/tables/activity_type.sql]: @joswig or @bashbash Can you comment here please?

**Review by mattdailis — APPROVED** [review]
Thanks for addressing my concerns :rocket: This PR has my conditional approval, pending an explicit decision on "subsystem" vs "category"

File: merlin-server/sql/merlin/tables/activity_type.sql
Comment (joswig) [reply merlin-server/sql/merlin/tables/activity_type.sql]: I'd like to go with the more generic name category, I'm good with limiting it to zero or one category for the initial implementation

Comment (Mythicaeda) [reply merlin-server/sql/merlin/tables/activity_type.sql]: I can make the change, but in answer to your original question, @mattdailis, ActivityTypes using `subsystem` was documented in [this Figma](https://www.figma.com/file/JsDIKVE0fR3mfiLYDJ01FM/AERIE-Metadata-Discussion-%23575?type=whiteboard&node-id=27-6010&t=7mvyRuLg7i5qh05f-0), in the part circled in blue on the right.

Comment (bashbash) [reply merlin-server/sql/merlin/tables/activity_type.sql]: It is indeed a category but I would call it sub-system, category is way too generic is my 2 cents. there may be other classification category besides sub-system

**Comment by joswig** [comment]
Theresa and I talked it out, and we converged on subsystem, since that aligns with Basak's thinking lets consider this settled.

**Pull request merged**
